src/drivers/mct/main/cime_comp_mod.F90

```diff
@@ -1323,6 +1323,7 @@ subroutine cime_init()
        write(logunit,F00) 'Initialize each component: atm, lnd, rof, ocn, ice, glc, wav, esp, iac'
        call shr_sys_flush(logunit)
     endif
+    call seq_infodata_GetData(infodata, cime_model=cime_model)
```

Comment:
pr #3536 already addresses this - I believe this should be removed.

Reply:
I thought that too, at first, but then I wondered about the timing of when `cime_model` is set vs. when it's needed. It's probably cleanest if this call is made in a single place fairly early in initialization.

Reply:
This should be removed, it is already called at line 3431 of this file.  @billsacks I don't understand your 'but'.

Reply:
Looking more closely at the calling sequence of various routines, I agree that I think things would work if this were removed. However, I'm uncomfortable only having this at line 3431, which is inside the routine `cime_printlogheader`: It happens right now that that routine is called before `cime_model` is needed, but it's very non-obvious that the setting of `cime_model` in that routine – which is about printing information – is critical for the correct operation of the code. What if someone reworked that print routine to no longer need `cime_model`, for example, or changed when it is called in initialization? In these cases, a seemingly innocent change would lead to a hard-to-understand bug.

My thinking was to avoid this problem by either:

1. Getting rid of both of the current calls to `seq_infodata_GetData(infodata, cime_model=cime_model)`, instead making this call as early as possible in either `cime_pre_init1` or `cime_pre_init2`, where it will be more obvious that it's needed.

2. Make `cime_model` a local variable in the subroutines that need it rather than a module-level variable, and calling `seq_infodata_GetData(infodata, cime_model=cime_model)` from each subroutine that needs this variable – because (as we saw from the bug that led to #3536 as well as the need for the addition here) the use of module-level data can be more error-prone.

I'm fine with either of the above solutions.

Reply:
@billsacks , done. I went with option 1 since it seemed easier.

Reply:
@jgfouca thanks, but as I look at the code, I'm wondering if the new location is now too early: It looks like the infodata object may not be initialized until the start of `cime_pre_init2`:

```Fortran
    !----------------------------------------------------------
    !| Initialize infodata
    !----------------------------------------------------------

    if (len_trim(cpl_inst_tag) > 0) then
       call seq_infodata_init(infodata,nlfilename, GLOID, pioid, &
            cpl_tag=cpl_inst_tag)
    else
       call seq_infodata_init(infodata,nlfilename, GLOID, pioid)
    end if
```

So I'm thinking that the new call may need to be after that.

Reply:
@billsacks , thanks. Restarting my testing.

Reply:
Looks good to me now, thanks.